### PR TITLE
Feature/improve backup restore ui

### PR DESCRIPTION
**Pull request by bart88**
- Set weights for the local tasks displayed on the site views. 
- Remove the custom operations views field plugin and just use the entity operations instead.
- Implement hook to provide alters to the entity operations and place logic in a service.

### CODE REVIEW

File: web/modules/custom/shp_backup/src/Service/Backup.php [hunk 1]
```diff
@@ -190,4 +195,30 @@ public function restore(NodeInterface $backup, NodeInterface $environment) {
...
 
+  /**
+   * Apply alterations to entity operations.
+   *
+   * @param array $operations
+   *   List of operations.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Entity to apply operations to.
+   */
+  public function operationsLinksAlter(array &$operations, EntityInterface $entity) {
```
Comment (pingers) [inline web/modules/custom/shp_backup/src/Service/Backup.php]: This looks a lot simpler. Nice one!

File: web/modules/custom/shp_backup/shp_backup.module [hunk 2]
```diff
@@ -112,3 +113,13 @@ function shp_backup_node_insert(NodeInterface $node) {
       break;
   }
 }
+
+function shp_backup_entity_operation_alter(array &$operations, EntityInterface $entity) {
+  // Get the current route name.
+  $route_name = \Drupal::routeMatch()->getRouteName();
+  // Operations are applied to the entity.
+  // To avoid issue, just bind them to this view.
+  if ($entity->getType() === "shp_environment" && $route_name === "view.shp_site_environments.page_1") {
```
Comment (pingers) [inline web/modules/custom/shp_backup/shp_backup.module]: Do we need to restrict to this view? I think we just want to restrict on entity type and allow permissions to deal with access to the operations.

File: web/modules/custom/shp_backup/src/Service/Backup.php [hunk 3]
```diff
@@ -190,4 +195,30 @@ public function restore(NodeInterface $backup, NodeInterface $environment) {
...
+    $site = $entity->field_shp_site->target_id;
+    $environment = $entity->id();
+
+    $operations['backup'] = [
```
Comment (pingers) [inline web/modules/custom/shp_backup/src/Service/Backup.php]: Wrap the operations in permissions - just like https://api.drupal.org/api/drupal/core%21modules%21field_ui%21field_ui.module/function/field_ui_entity_operation/8.3.x

File: web/modules/custom/shp_backup/shp_backup.module (hunk 2 above)
Comment (caseyfw) [reply web/modules/custom/shp_backup/shp_backup.module]: Agreed.

File: web/modules/custom/shp_custom/config/install/views.view.shp_site_environments.yml [hunk 4]
```diff
@@ -277,64 +277,14 @@ display:
...
-          plugin_id: shp_backup_operations
         operations:
           id: operations
           table: node
           field: operations
           relationship: none
           group_type: group
           admin_label: ''
-          label: ''
+          label: Operations
```
Comment (caseyfw) [inline web/modules/custom/shp_custom/config/install/views.view.shp_site_environments.yml]: Nice. I thought it was weird that an operation column added to a view has no label by default.

**Pull request merged**